### PR TITLE
[cloud-provider-aws] Add ability to configure additional policies for IAM roles

### LINKED ISSUE #1005: [cloud-provider-aws] Add ability to configure additional policies for IAM roles
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

There are several cases when you might need to give access to AWS API for applications running on nodes. We have a basic role that is attached to every node in the cluster. Additional policies can be attached to it.

### Proposed Solution

Add additionalRolePolicies parameter in AWSClusterConfiguration.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by gmaiorov**
## Description

Add an `additionalRolePolicies` parameter to AWSClusterConfiguration and update.
## Why do we need it, and what problem does it solve?

Fixes #1005. 

## What is the expected result?

Following changes are expected:
- `additionalRolePolicies` parameter added to AWSClusterConfiguration;
- corresponding policies attached to the [`Action`](https://github.com/deckhouse/deckhouse/blob/498eb618bbdd98b21950b1795eb02b6e2a3e50b2/candi/cloud-providers/aws/layouts/standard/base-infrastructure/main.tf#L165)  list in `aws_iam_role_policy` resource.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: feature
summary: Add an `additionalRolePolicies` parameter to AWSClusterConfiguration (#1005)
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @gmaiorov. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @gmaiorov. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @gmaiorov. I've started the workflow for label 'e2e/run/aws'...

**Comment by gmaiorov** [comment]
[Smoke test report PR2256.pdf](https://github.com/deckhouse/deckhouse/files/9401126/Smoke.test.report.PR2256.pdf)

**Review by RomanenkoDenys — APPROVED** [review]

File: candi/cloud-providers/aws/layouts/standard/base-infrastructure/main.tf [hunk 1]
```diff
@@ -163,6 +163,9 @@ resource "aws_iam_role_policy" "node" {
       {
         "Effect": "Allow",
         "Action": [
+          %{ for policy in local.additional_role_policies }
```
Comment (konstantin-axenov) [inline candi/cloud-providers/aws/layouts/standard/base-infrastructure/main.tf:166]:
```suggestion
          %{for policy in local.additional_role_policies}
```

File: candi/cloud-providers/aws/layouts/with-nat/base-infrastructure/main.tf [hunk 2]
```diff
@@ -163,6 +163,9 @@ resource "aws_iam_role_policy" "node" {
       {
         "Effect": "Allow",
         "Action": [
+          %{ for policy in local.additional_role_policies }
```
Comment (konstantin-axenov) [inline candi/cloud-providers/aws/layouts/with-nat/base-infrastructure/main.tf:166]:
```suggestion
          %{for policy in local.additional_role_policies}
```

File: candi/cloud-providers/aws/layouts/with-nat/base-infrastructure/main.tf [hunk 3]
```diff
@@ -163,6 +163,9 @@ resource "aws_iam_role_policy" "node" {
...
+          "${policy}",
+          %{ endfor }
```
Comment (konstantin-axenov) [inline candi/cloud-providers/aws/layouts/with-nat/base-infrastructure/main.tf:168]:
```suggestion
          %{endfor}
```

File: candi/cloud-providers/aws/layouts/without-nat/base-infrastructure/main.tf [hunk 4]
```diff
@@ -112,6 +112,9 @@ resource "aws_iam_role_policy" "node" {
       {
         "Effect": "Allow",
         "Action": [
+          %{ for policy in local.additional_role_policies }
```
Comment (konstantin-axenov) [inline candi/cloud-providers/aws/layouts/without-nat/base-infrastructure/main.tf:115]:
```suggestion
          %{for policy in local.additional_role_policies}
```

File: candi/cloud-providers/aws/layouts/standard/base-infrastructure/main.tf (hunk 1 above)
Comment (konstantin-axenov) [reply candi/cloud-providers/aws/layouts/standard/base-infrastructure/main.tf:166]: These changes of terraform files are just to match `terraform fmt`. Not all existing files are correctly formatted, but if we add or change code, it's better to format it correctly.

**Review by konstantin-axenov — CHANGES_REQUESTED** [review]

File: candi/cloud-providers/aws/openapi/cluster_configuration.yaml [hunk 5]
```diff
@@ -343,6 +343,13 @@ apiVersions:
         minItems: 1
         uniqueItems: true
         description: The globally restricted set of zones that this cloud provider works with.
+      additionalRolePolicies:
+        type: array
+        items:
+          type: string
+          pattern: '^(\w+):(\w+)$'
+        minItems: 1
+        description: List containing an additional policies for IAM roles.
```
[the remark this answers is not in the document]
Comment (gmaiorov) [reply candi/cloud-providers/aws/openapi/cluster_configuration.yaml]: I've clarified additionalRolePolicies parameter documentation and added a translated description in 746812d9085c0bf4087bd6c0efad2d3154dcb03e.

**Comment by gmaiorov** [comment]
Rebased onto main & force-pushed to fix matrix tests.

**Review by konstantin-axenov — APPROVED** [review]

**Review by konstantin-axenov — APPROVED** [review]

**Pull request merged**